src/bio.c

```diff
@@ -220,10 +220,16 @@ void *bioProcessBackgroundJobs(void *arg) {
         if (type == BIO_CLOSE_FILE) {
             close(job->fd);
         } else if (type == BIO_AOF_FSYNC) {
-            if (redis_fsync(job->fd) == -1) {
+            /* The fd may be closed by main thread and reused for another
+             * socket, pipe, or file. We just ignore these errno because
+             * aof fsync did not really fail. */
+            if (redis_fsync(job->fd) == -1 &&
+                (errno != EBADF || errno != EINVAL))
```

Comment:
mixup between `||` and `&&`
```suggestion
            if (redis_fsync(job->fd) == -1 &&
                errno != EBADF && errno != EINVAL)
```

Reply:
oh, my bad, thanks